forms/MemberDatetimeOptionsetField.php

```diff
@@ -5,65 +5,69 @@
-			if($key == $this->value) {
-				$useValue = false;
-				$checked = " checked=\"checked\"";
-			} else {
-				$checked = "";
+		// Add all options striped
+		$anySelected = false;
+		foreach($this->getSourceEmpty() as $value => $title) {
+			$odd = !$odd;
+			if(!$anySelected) {
+				$anySelected = $this->isSelectedValue($value, $this->Value());
 			}
+			$options[] = $this->getFieldOption($value, $title, $odd);
+		}
 
-			$odd = ($odd + 1) % 2;
-			$extraClass = $odd ? "odd" : "even";
-			$extraClass .= " val" . preg_replace('/[^a-zA-Z0-9\-\_]/', '_', $key);
-			$disabled = ($this->disabled || in_array($key, $this->disabledItems)) ? "disabled=\"disabled\"" : "";
-			$ATT_key = Convert::raw2att($key);
+		// Add "custom" input field option
+		$options[] = $this->getCustomFieldOption(!$anySelected, !$odd);
 
-			$options .= "<li class=\"".$extraClass."\">"
-				. "<input id=\"$itemID\" name=\"$this->name\" type=\"radio\" value=\"$key\"$checked $disabled"
-				. " class=\"radio\" /> <label title=\"$ATT_key\" for=\"$itemID\">$value</label></li>\n";
-		}
+		// Build fieldset
+		$properties = array_merge($properties, array(
+			'Options' => new ArrayList($options)
+		));
 
+		return $this->customise($properties)->renderWith(
+			$this->getTemplates()
+		);
+	}
+
+	/**
+	 * Create the "custom" selection field option
+	 *
+	 * @param bool $isChecked True if this is checked
+	 * @param bool $odd Is odd striped
+	 * @return ArrayData
+	 */
+	protected function getCustomFieldOption($isChecked, $odd) {
 		// Add "custom" input field
-		$value = ($this->value && !array_key_exists($this->value, $this->source)) ? $this->value : null;
-		$checked = ($value) ? " checked=\"checked\"" : '';
-		$options .= "<li class=\"valCustom\">"
-			. sprintf(
-					"<input id=\"%s_custom\" name=\"%s\" type=\"radio\" value=\"__custom__\" class=\"radio\" %s />",
-					$itemID, $this->name,
-					$checked
-				)
-			. sprintf(
-					'<label for="%s_custom">%s:</label>',
-					$itemID, _t('MemberDatetimeOptionsetField.Custom', 'Custom')
-				)
-			. sprintf(
-					"<input class=\"customFormat cms-help cms-help-tooltip\" name=\"%s_custom\" value=\"%s\" />\n",
-				$this->name, Convert::raw2xml($value)
-			)
-			. sprintf(
-				"<input type=\"hidden\" class=\"formatValidationURL\" value=\"%s\" />",
-				$this->Link() . '/validate'
-			);
-		$options .= ($value) ? sprintf(
-			'<span class="preview">(%s: "%s")</span>',
-			_t('MemberDatetimeOptionsetField.Preview', 'Preview'),
-			Convert::raw2xml(Zend_Date::now()->toString($value))
-		) : '';
+		$option = $this->getFieldOption(
+			self::CUSTOM_OPTION,
+			_t('MemberDatetimeOptionsetField.Custom', 'Custom'),
+			$odd
+		);
+		$option->setField('isChecked', $isChecked);
+		$option->setField('CustomName', $this->getName().'[Custom]');
+		$option->setField('CustomValue', $this->Value());
+		if($this->Value()) {
+			$preview = Convert::raw2xml(Zend_Date::now()->toString($this->Value()));
```

Comment:
We shouldn't use "now" as the date string because it can be ambiguous. We should show a non-ambiguous date string for users to pick from. eg: 25/12/2015 (maybe the year can be current!?)